ament_cmake_target_dependencies/cmake/ament_get_recursive_properties.cmake

```diff
@@ -0,0 +1,76 @@
+# Copyright 2020 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# Get the recrusive include directories and libraries from interface targets.
+#
+# :param var_include_dirs: the output variable name
+# :type var_include_dirs: string
+# :param var_libraries: the output variable name
+# :type var_libraries: string
+# :param ARGN: a list of interface targets
+# :type ARGN: list of strings
+#
+# @public
+#
+function(ament_get_recursive_properties var_include_dirs var_libraries)
+  set(all_include_dirs "")
+  set(all_libraries "")
+
+  if(${ARGC} GREATER 0)
+    foreach(target ${ARGN})
+      # only use actual targets
+      if(NOT TARGET "${target}")
+        continue()
```

Comment:
We should probably add a warning or error message here so the user knows they've made a mistake.

Reply:
In other cases we use it under the same condition since some packages add other information to that variable (which are not targets):
* https://github.com/ament/ament_cmake/blob/master/ament_cmake_export_dependencies/cmake/ament_cmake_export_dependencies-extras.cmake.in#L29-L55
* https://github.com/ament/ament_cmake/blob/bae0fb8828f5cfe16f47dd9587762a26de6bac9f/ament_cmake_target_dependencies/cmake/ament_target_dependencies.cmake#L74-L89